analyses/cell-type-wilms-tumor-06/00_fetal_reference_kidney.Rmd

```diff
@@ -0,0 +1,240 @@
+  refDR = "pca",
+  refAssay = "SCT",
+  dims = 1:50,
+  k.param = 31,
+  plotref = "umap",
+  plot.metadata = NULL,
+  ori.index = NULL,
+  colormap = NULL,
+  assays = NULL,
+  metadata = c("compartment", "cell_type"),
+  reference.version = "0.0.0",
+  verbose = FALSE
+)
+
+
+d2 <- do_DimPlot(Fetal_kidney, reduction = "refUMAP", dims = c(1,2), group.by = "compartment", label = TRUE, repel = TRUE) + NoLegend() + ggtitle("umap reduction after SCTransform and Seurat workflow")
+d3 <- do_DimPlot(Fetal_kidney, reduction = "refUMAP", dims = c(1,2), group.by = "cell_type", label = TRUE, repel = TRUE) + NoLegend()
+
+d2 | d3
+
+# save reference in rds file
+SaveAnnoyIndex(object = Fetal_kidney[["refdr.annoy.neighbors"]], file = file.path(path_to_output, "idx.annoy"))
+saveRDS(object = Fetal_kidney, file = file.path(path_to_output, "ref.Rds"))
+
+# finally, we can remove the fetal_full.rds file from the marker-sets directory
+file.remove(path_to_data)
+```
+## Characterization of compartment and cell types in the reference
+
+Here, we use an unbiased approach to find transcripts that characterized the different compartments and cell types.
+
+This is just to get markers genes of the different population, in case some could be of interest for the Wilms tumor annotations. 
+
+We run DElegate::FindAllMarkers2 to find markers of the different clusters and manually check if they do make sense. 
+DElegate::FindAllMarkers2 is an improved version of Seurat::FindAllMarkers based on pseudobulk differential expression method. 
+Please check the preprint from Chistoph Hafemeister: https://www.biorxiv.org/content/10.1101/2023.03.28.534443v1
+and tool described here: https://github.com/cancerbits/DElegate 
+
+### Find marker genes for each of the compartment
+
+
+```{r markers_compatment, fig.width=8, fig.height=7, out.width='100%'}
+de_results   <- DElegate::FindAllMarkers2(s, group_column = "compartment")
+
+#filter the most relevant markers
+s.markers <- de_results[de_results$padj < cfg$padj_thershold & de_results$log_fc > cfg$lfc_threshold & de_results$rate1 > cfg$rate1_threshold,]
+
+
+DT::datatable(s.markers, caption = ("marker genes"), 
+              extensions = 'Buttons', 
+              options = list(  dom = 'Bfrtip',
+
+                               buttons = c( 'csv', 'excel')))
+
+# Select top 5 genes for heatmap plotting
+s.markers <- na.omit(s.markers)
+s.markers %>%
+    group_by(group1) %>%
+    top_n(n =  5, wt = log_fc) -> top5
+
+# subset for plotting
+Idents(s) <- s$compartment
+cells <- WhichCells(s, downsample = 100)
+ss <- subset(s, cells = cells)
+ss <- ScaleData(ss, features = top5$feature)
+
+p1 <- SCpubr::do_DimPlot(s, reduction="umap", group.by = "compartment", label = TRUE, repel = TRUE) + ggtitle("compartment")
+p2 <- DoHeatmap(ss, features = top5$feature,  cells = cells, group.by = "compartment") + NoLegend() + 
+  scale_fill_gradientn(colors =  c("#01665e","#35978f",'darkslategray3', "#f7f7f7", "#fee391","#fec44f","#F9AD03")) 
```

Comment:
I think this is the same color palette as what is used below? So I'd recommend saving it as a variable so you only would need to change it one place:

```r
heatmap_color_palette <- c("#01665e","#35978f",'darkslategray3', "#f7f7f7", "#fee391","#fec44f","#F9AD03")
```